DashAI/back/api/api_v1/endpoints/datasets.py

```diff
@@ -58,47 +64,89 @@ def parse_params(params):
 
 
 @router.get("/")
-async def get_dataset():
+async def get_datasets():
+    """
+    Returns all the available datasets in the database.
+
+    Returns
+    -------
+    List[JSON]
+        List of dataset JSONs
+    """
+
+    try:
+        all_datasets = session.query(models.Dataset).all()
+        if not all_datasets:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail="No datasets found"
+            )
```

Comment:
What if you simply return an empty array?

Reply:
There are [mixed](https://stackoverflow.com/questions/11746894/what-is-the-proper-rest-response-code-for-a-valid-request-but-an-empty-data) opinions on this topic. But most comments are in favour of 404 when you query an unexisting object